esp-wifi/src/compat/malloc.rs

```diff
@@ -2,7 +2,7 @@ use core::alloc::Layout;
 
 use crate::HEAP;
 
-pub unsafe extern "C" fn malloc(size: usize) -> *const u8 {
+pub unsafe extern "C" fn malloc(size: usize) -> *mut u8 {
```

Comment:
Nice find, I'm pretty certain this is instant-UB if we ever tried to mutate it (which we almost certainly did, I mean who allocates memory just to let it sit there?)

Reply:
I'm not sure about UB. Rust seems to be a bit more relaxed around pointer than references. There is already a mut -> const cast here anyway behind the scenes.